packages/permissions/src/utils/has-permissions.ts

```diff
@@ -270,16 +318,25 @@ export const hasAppliedDataFence = (options: TOptionsForAppliedDataFence) => {
       if (!specificActualDataFence) return false;
 
       const [dataFenceName, dataFenceValue] = specificActualDataFence;
-      return hasDemandedDataFence({
-        actualDataFence: {
-          name: dataFenceName,
-          // we do the type casting because at this point we are sure that
-          // specificActualDataFence.dataFenceValue is not null
-          dataFenceValue: dataFenceValue as TActualDataFence['dataFenceValue'],
-        },
+      const actualDataFence = {
+        name: dataFenceName,
+        // we do the type casting because at this point we are sure that
+        // specificActualDataFence.dataFenceValue is not null
+        dataFenceValue: dataFenceValue as TActualDataFence['dataFenceValue'],
+      };
+
+      const isPermissionOverwritingDataFence = getIsPermissionOverwritingDataFence(
+        options.actualPermissions,
+        demandedDataFence
+      );
+
+      const hasDemandedDataFence = getHasDemandedDataFence({
+        actualDataFence,
         demandedDataFence,
         selectDataFenceData: options.selectDataFenceData,
       });
+
+      return !isPermissionOverwritingDataFence && hasDemandedDataFence;
```

Comment:
Note, we return false here now in some cases. This makes the `useIsAuthorized` fall through to the resource access and action right evaluation.